### PR TITLE
Upgrade SQLite to fix Xcode 10.2 errors (#211)

### PR DESCRIPTION
**Pull request by palpatim**
- Updated SQLite.swift to specific master branch commit instead of official tagged release
- Added unit tests for #122
- Fixed whitespace linting violations
- Import AppSyncLogHelper instead of CocoaLumberjack in AppSync Objective-C
  classes
- Added SQLite to TestCommon dependencies to allow for direct DB access for
  some setup unit tests
- Updated AWS dependencies to 2.9.5
- Moved long-running performance tests to integration test suite to allow for
  rapid execution of unit tests during development

*Issue #, if available:* #211 

*Description of changes:*

This is a large diff because it includes AWS dependency pod updates. Meaningful changes are:

* **Core code**
  * Import `AppSyncLogHelper` instead of `CocoaLumberjack` in Objective-C code, which was required by the update to AWS SDK 2.9.5
* **Podfile**
  * Updated SQLite dependency to master
  * Added targets for test host app
  * Added SQLite as a TestCommon dependency to allow direct DB access in unit tests
* **Test code**
  * Added unit test for #211. The test includes direct DB access, but that seemed to be the most appropriate way to validate the low-level dependency. Note that the test passes in Xcode 10.1 *without* upgrading SQLite.swift, so if you want to validate the test behavior, be sure to use Xcode 10.2 before reverting the SQLite.swift version
  * Moved some long-running performance & unit tests to the integration test bundle to make unit tests faster to run during development
* **Dependencies**
  * Updated SQLite.swift

I've reviewed the SQLite.swift [diff between 0.11.5 and master](https://github.com/stephencelis/SQLite.swift/compare/0.11.5...master) and do not see any changes that give me pause. The change to remove the dependency on `#function` is far-reaching, and I'm unclear on that library's test coverage, so it's possible there are cases that aren't properly handled, but all our tests pass and my spot checks of caching behavior seem to bear this out.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,18 @@
 # AWS AppSync SDK for iOS - CHANGELOG
 
-The AWS AppSync SDK for iOS enables you to access your AWS AppSync backend and perform operations like `Queries`, `Mutations` and `Subscriptions`. The SDK also includes support for offline operations.
+The AWS AppSync SDK for iOS enables you to access your AWS AppSync backend and perform operations like `Queries`, `Mutations` and `Subscriptions`. The SDK
+also includes support for offline operations.
```
Comment (rohandubal) [inline CHANGELOG.md]: I think there is a line break inserted in error?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+
+## 2.11.1
+
+### Bug Fixes
+- Updated SQLite.swift to master to fix caching error. ([Issue #211](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/issues/211))
+
+### Misc. Updates
+- Updated AWS SDK dependencies to 2.9.5
+- Updated SQLite.swift to the latest version of the master branch instead of official release. The latest official release of SQLite.swift does not work
+  with apps compiled in Xcode 10.2 because it relies on the format of the `#function` expression behavior, which changed between 10.1 and 10.2. This
```
Comment (rohandubal) [inline CHANGELOG.md]: Are the line breaks caused due to some IDE setting? Not sure of the rendering due to new lines.

File: CHANGELOG.md (hunk 1 above)
Comment (palpatim) [reply CHANGELOG.md]: That's intentional, I'm inserting hard line breaks into the long lines of text to make reviews & editing easier in text editors like vi.

File: CHANGELOG.md (hunk 2 above)
Comment (palpatim) [reply CHANGELOG.md]: That's intentional, I'm inserting hard line breaks into the long lines of text to make reviews & editing easier in text editors like vi.

**Pull request merged**